modules/diff.js

```diff
@@ -168,32 +175,26 @@ const worker = async ({ product, response_url }) => {
 }
 
 const route = (req, res) => {
-  const { text: product = 'firstorder', response_url } = req.body || {}
-
+  const { user_id, text: product = 'firstorder', response_url } = req.body // extract payload from slash command
   const payload = { product, response_url }
+  const { groups = [] } = SERVICES[product] || CLIENTS[product] || {}
 
-  if (DEPLOYED) {
-    lambda.invoke({
-      FunctionName: getFuncName('slack'),
-      InvocationType: 'Event',
-      Payload: JSON.stringify({ type: 'diff', payload }),
-    }, (err) => {
-      if (err) {
-        console.error(err)
-        return res.status(200).json({ response_type: 'ephemeral', text: 'Failed to diff' })
-      }
-      return res.status(200).json({
-        response_type: 'ephemeral',
-        text: `Diffing for ${product}...`,
-      })
-    })
-  } else {
-    worker(payload).catch(console.error)
-    return res.status(200).json({
-      response_type: 'ephemeral',
-      text: `Diffing for ${product}...`,
-    })
-  }
+  return userInGroup({ user_id, groups }).then((can) => {
+    if (!can) {
+      return res.status(200).json({ response_type: 'ephemeral', text: `You cannot diff ${product}` })
```

Comment:
why default is FO? when I try it says I can't diff - cause I might not be in snoke or fo team. Not that this is an issue - can be added, or not - but more like the message should be towards  `you should pick a repo/product to diff`

Reply:
yea it's more of a "backward compatibility" thing as the original implementation defaulted to FO. can change in the future.

you are in both of those user groups though, so the error might be something else.

Reply:
You're right. When I do `/diff firstorder` it's all good. Maybe because when I do the command without params, the `test` is being passed as blank `''` and overriding the default instead of not being passed at all as expected (?)

Reply:
yea probably the blank string case.